build/v1/types.go

```diff
@@ -1016,6 +1022,29 @@ type ImageChangeTrigger struct {
 	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
 }
 
+// ImageChangeTriggerStatus tracks the latest resolved status of the associated ImageChangeTrigger policy
+// specified in the BuildConfigSpec.Triggers struct.
+type ImageChangeTriggerStatus struct {
+	// lastTriggeredImageID contains the resolved DockerImage sha from the ImageStreamTag referenced in the
+	// ImageChangeTrigger.From field when that trigger resulted in a Build being initiated.
+	LastTriggeredImageID string `json:"lastTriggeredImageID,omitempty" protobuf:"bytes,1,opt,name=lastTriggeredImageID"`
+
+	// from is whichever ImageStreamTag, either an explicitly cited ImageChangeTrigger ImageStreamTag in the spec,
+	// or the From setting in the build strategy if the From field in the ImageChangeTrigger is not set, that resulted in
+	// the triggering of a build.  This field can be used to correlate with specific ImageChangeTrigger policicy specifications
+	// in the BuildConfig spec.
+	From *corev1.ObjectReference `json:"from,omitempty" protobuf:"bytes,2,opt,name=from"`
+
+	// paused is true if this trigger is temporarily disabled, and the setting on the spec has been reconciled. Optional.
+	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
+
+	// lastTriggerTime is the last time the BuildConfig was triggered by an ImageStreamTag change.
+	LastTriggerTime metav1.Time `json:"lastTriggerTime,omitempty" protobuf:"bytes,4,opt,name=lastTriggerTime"`
```

Comment:
i'll admit this and LastPausedTime are minimally flirting with left field :-) 

I almost consolidated both of this into LastUpdateTime (where it does not matter whether a trigger or pause change lead to the update), but I feft it as is for now

and certainly a time stamp in general here may not be a slam dunk choice, but it felt like potentially useful metadata for debug / diagnostics

Reply:
lastpausedtime isn't about when the build was triggered, it's about when the trigger was paused, so consolidating them wouldn't be the right thing to do.

i think the idea is to be able to understand "when did this trigger get disabled".  Which, yeah, should be in status and not spec.  So if we're going to keep having it, putting it here makes sense to me.

Reply:
Should we adopt a similar convention to upstream conditions? 

PausedStatus
LastPausedTransitionTime